### DISCUSSION TITLE
Using build artifacts from netbeans for personal github actions

### DISCUSSION THREAD

**Discussion by haidubogdan** [apache/netbeans#7321] [Q&A] [answered]
Hi, I'm not sure who can answer this.

I see that the free github account has a free storage limit of 500MB.

I wonder if we should be careful when downloading builds from apache/netbeans repo github actions ?

I don't want to increase the quota cost for this repository.

It's more part of an experiment to be able to build npm plugins directly with github actions.

**Comment by mbien** [comment]
> I wonder if we should be careful when downloading builds from apache/netbeans repo github actions ?
> I don't want to increase the quota cost for this repository.

I am not sure what you mean by "this repository". NetBeans release artifacts are released to maven central (and the IDE packages to the download page), the typical way to depend on NB would be to simply add maven dependencies (e.g for plugins or NB platform applications).

> I wonder if we should be careful when downloading builds from apache/netbeans repo github actions ?

what would you download from apache/netbeans repo github actions? I am a bit confused ;)

> I don't want to increase the quota cost for this repository.
> It's more part of an experiment to be able to build npm plugins directly with github actions.

according to https://github.com/settings/billing/summary, quote:

> Storage for Actions and Packages
> Shared storage consists of Actions artifacts and Packages usage. This graph shows the account's storage usage in GB-months. Removing stored artifacts will not reduce this number, but it will lower its rate of growth. To see your account's current storage, download a usage report.

gh action artifacts and release packages influence storage. So it will depend what you actually do in your gh actions. (also keep an eye on the CI usage minutes)

**Comment by haidubogdan** [comment]
I was thinking when someone would access this kind of links : 
![image](https://github.com/apache/netbeans/assets/15365717/b658e933-a22d-4a19-83f3-7735147c83db)

https://github.com/apache/netbeans/actions/runs/8803272227/artifacts/1440047180

I was going on a presumption that getting artifacts directly from github would be faster.

Now I also remembered that we have build links  https://ci-builds.apache.org/job/Netbeans/job/netbeans-linux/lastSuccessfulBuild/artifact/nbbuild/ 

So I will start my experiments by using the official build website.

The end goal is to experiment with a deploy + build **nbm** system for a plugin.

Thank you for your answer.

**Reply by mbien** [reply, marked as answer]
The build artifact is a temporary artifact of the workflow and cleaned up once all jobs finished. You can't download that unless the build fails. This is there for debugging purposes.

The dev builds on jenkins are for manual testing of the most recent bits directly from the master branch.

But again. you should not download netbeans at all during the build. plugins would simply depend on some modules and thats it.
example:
https://github.com/mbien/xpath-util
https://github.com/mbien/xpath-util/blob/master/pom.xml

**Reply by haidubogdan** [reply]
I'm used to using `ant` to build **nbm** plugins and that is dependent on the `netbeans` build files.
But I will have to consider / learn using `maven` for this type of task :) .

**Reply by haidubogdan** [reply]
In the end I've managed to have a nbm packaging github actions +maven, so thanks for your direction :) .
Not a lot of documentation though for maven / netbeans plugin packaging.

I will leave here a quick **nbm** packaging pipeline  from ant based project using maven : 
https://github.com/haidubogdan/netbeans-php-blade-plugin/blob/main/.github/workflows/maven.yml

For the moment it's missing a script  to use  the api version from nbproject/project.properties to set the pom.xml package version.
